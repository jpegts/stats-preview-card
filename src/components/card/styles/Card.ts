import styled from "styled-components/macro";
import { media } from "../../../styles/global-style";
import {
  neutralAccent,
  neutralMain,
  neutralSecond,
  primaryAccent,
  primaryMain,
  primarySecond,
} from "../../../styles/style-guide";

export const Section = styled.section`
  min-height: 100vh;
  background-color: ${primaryMain};
  display: grid;
  place-items: center;
  color: ${neutralMain};
`;

export const Container = styled.div`
  background-color: ${primarySecond};
  width: 325px;
  border-radius: 8px;
  height: 780px;
  display: flex;
  flex-direction: column-reverse;
  justify-content: flex-end;
  overflow: hidden;

  ${media.desktop} {
    height: 446px;
    min-width: 1110px;
    display: flex;
    flex-direction: row;
  }
`;

export const Pane = styled.div`
  text-align: center;

  &:first-of-type {
    padding: 24px;
  }

  ${media.desktop} {
    text-align: left;
    &:first-of-type {
      padding: 25px;
      width: 982px;
    }
    &:last-of-type {
      width: 972px;
    }
  }
`;

export const Picture = styled.div`
  background-image: url("/images/image-header-mobile.jpg");
  filter: brightness(33%) sepia(100%) hue-rotate(-120deg) saturate(580%)
    contrast(0.8);
  height: 239px;
  background-size: cover;

  ${media.desktop} {
    background-image: url("/images/image-header-desktop.jpg");
    height: 100%;
  }
`;

export const Title = styled.h3`
  font-size: 28px;
  font-family: "Inter", sans-serif;
  margin: 16px auto;

  & > span {
    color: ${primaryAccent};
  }

  ${media.desktop} {
    padding-left: 46px;
    padding-top: 32px;
    padding-right: 20px;
    font-size: 36px;
    line-height: 43px;
  }
`;

export const Description = styled.p`
  color: ${neutralSecond};
  font-family: "Inter";
  margin: 18px 10px;
  font-size: 15px;
  line-height: 24px;

  ${media.desktop} {
    margin-right: 0;
    margin-left: 0;
    padding-left: 46px;
    padding-top: 8px;
    padding-right: 80px;
  }
`;

export const Group = styled.div`
  padding-top: 25px;
  ${media.desktop} {
    display: flex;
    padding-left: 46px;
    padding-top: 57px;
  }
`;

export const Item = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  margin: 0 auto;
  margin-bottom: 22px;
  width: 80px;

  ${media.desktop} {
    align-items: flex-start;
    margin-right: 64px;
    margin-left: 0px;
  }
`;

export const Stat = styled.p`
  font-weight: bold;
  font-family: "Inter";
  font-size: 24px;
  margin: 0;
`;

export const Meta = styled.p`
  color: ${neutralAccent};
  text-transform: uppercase;
  letter-spacing: 0.5px;
  margin: 8.5px 0px;
  font-family: "Inter";
  font-size: 13px;

  ${media.desktop} {
    font-family: "Lexend Deca";
    align-items: flex-start;
    margin-right: 64px;
  }
`;
