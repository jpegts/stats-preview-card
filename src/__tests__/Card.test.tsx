import React from "react";
import { render } from "@testing-library/react";
import Card from "../components/card";

describe("<Card />", () => {
  it("renders <Card/> with populated data", () => {
    const { container, getByText, getByTestId } = render(
      <Card.Section>
        <Card>
          <Card.Pane>
            <Card.Title data-testid='title-test'>
              Get <span>insights</span> that help your business grow.
            </Card.Title>
            <Card.Description>
              Discover the benefits of data analytics and make better decisions
              regarding revenue, customer experience, and overall efficiency.
            </Card.Description>
            <Card.Group>
              <Card.Item>
                <Card.Stat>10k+</Card.Stat>
                <Card.Meta>companies</Card.Meta>
              </Card.Item>
              <Card.Item>
                <Card.Stat>314</Card.Stat>
                <Card.Meta>templates</Card.Meta>
              </Card.Item>
              <Card.Item>
                <Card.Stat>12M+</Card.Stat>
                <Card.Meta>queries</Card.Meta>
              </Card.Item>
            </Card.Group>
          </Card.Pane>
          <Card.Pane>
            <Card.Picture />
          </Card.Pane>
        </Card>
      </Card.Section>
    );

    expect(
      getByText(
        "Discover the benefits of data analytics and make better decisions regarding revenue, customer experience, and overall efficiency."
      )
    ).toBeTruthy();
    expect(getByTestId("title-test")).toBeTruthy();
    expect(getByText("10k+")).toBeTruthy();
    expect(getByText("companies")).toBeTruthy();
    expect(getByText("314")).toBeTruthy();
    expect(getByText("templates")).toBeTruthy();
    expect(getByText("12M+")).toBeTruthy();
    expect(getByText("queries")).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
