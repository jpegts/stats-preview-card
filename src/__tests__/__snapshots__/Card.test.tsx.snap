// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> renders <Card/> with populated data 1`] = `
<section
  class="Card__Section-sc-1orjhe5-0 cgLRro"
>
  <div
    class="Card__Container-sc-1orjhe5-1 ffJUiR"
  >
    <div
      class="Card__Pane-sc-1orjhe5-2 hDJgGX"
    >
      <h3
        class="Card__Title-sc-1orjhe5-4 dSeTZi"
        data-testid="title-test"
      >
        Get 
        <span>
          insights
        </span>
         that help your business grow.
      </h3>
      <p
        class="Card__Description-sc-1orjhe5-5 kgpQhF"
      >
        Discover the benefits of data analytics and make better decisions regarding revenue, customer experience, and overall efficiency.
      </p>
      <div
        class="Card__Group-sc-1orjhe5-6 fPQjvw"
      >
        <div
          class="Card__Item-sc-1orjhe5-7 fFNmWk"
        >
          <p
            class="Card__Stat-sc-1orjhe5-8 fuYUOY"
          >
            10k+
          </p>
          <p
            class="Card__Meta-sc-1orjhe5-9 ljaYdK"
          >
            companies
          </p>
        </div>
        <div
          class="Card__Item-sc-1orjhe5-7 fFNmWk"
        >
          <p
            class="Card__Stat-sc-1orjhe5-8 fuYUOY"
          >
            314
          </p>
          <p
            class="Card__Meta-sc-1orjhe5-9 ljaYdK"
          >
            templates
          </p>
        </div>
        <div
          class="Card__Item-sc-1orjhe5-7 fFNmWk"
        >
          <p
            class="Card__Stat-sc-1orjhe5-8 fuYUOY"
          >
            12M+
          </p>
          <p
            class="Card__Meta-sc-1orjhe5-9 ljaYdK"
          >
            queries
          </p>
        </div>
      </div>
    </div>
    <div
      class="Card__Pane-sc-1orjhe5-2 hDJgGX"
    >
      <div
        class="Card__Picture-sc-1orjhe5-3 dGmdkY"
      />
    </div>
  </div>
</section>
`;
